simplq/src/components/pages/Home/MyQueues.jsx

```diff
@@ -1,41 +1,39 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useCallback } from 'react';
 import DeleteIcon from '@material-ui/icons/Delete';
 import IconButton from '@material-ui/core/IconButton';
 import { useHistory } from 'react-router';
-import { useAuth0 } from '@auth0/auth0-react';
+import { useFetchQueues, useDeleteQueue, selectQueues } from 'store/queues';
+import { useDispatch, useSelector } from 'react-redux';
 import styles from './home.module.scss';
-import { QueueRequestFactory } from '../../../api/requestFactory';
-import useRequest from '../../../api/useRequest';
 
 export default () => {
   const history = useHistory();
-  const { requestMaker } = useRequest();
-  const [myQueues, setMyQueues] = useState([]);
-  const { isAuthenticated } = useAuth0();
+  const dispatch = useDispatch();
+  const queues = useSelector(selectQueues);
+  const fetchQueues = useCallback(useFetchQueues(), []);
+  const deleteQueue = useDeleteQueue();
 
   useEffect(() => {
-    if (isAuthenticated)
-      requestMaker(QueueRequestFactory.getMyQueues()).then((resp) => setMyQueues(resp.queues));
-  }, [requestMaker, isAuthenticated]);
-
-  if (!isAuthenticated) {
-    return null;
-  }
+    dispatch(fetchQueues());
+  }, [dispatch, fetchQueues]);
 
   const handleDelete = (e, queue) => {
     // Don't trigger parent's onClick
+    const arg = { queueId: queue.queueId };
     e.stopPropagation();
-    requestMaker(QueueRequestFactory.deleteQueue(queue.queueId)).then(() => history.push('/'));
+    dispatch(deleteQueue(arg))
+      // waht is the next line?
+      .then(() => history.push('/'));
```

Comment:
Once queue is deleted, we take the user back to home page. In the long term, we would show admin page for deleted queues as well, but we won't need it in near future.

Reply:
The user is already on the home page. No need for this, will be removed.